cmd/admin-service-restart.go

```diff
@@ -119,20 +123,36 @@ func mainAdminServiceRestart(ctx *cli.Context) error {
 	// Success..
 	printMsg(serviceRestartCommand{Status: "success", ServerURL: aliasedURL})
 
-	// Max. time taken by the server to shutdown is 5 seconds.
-	// This can happen when there are lot of s3 requests pending when the server
-	// receives a restart command.
-	// Sleep for 6 seconds and then check if the server is online.
-	time.Sleep(6 * time.Second)
+	coloring := color.New(color.FgRed)
+	mark := "."
 
-	// Fetch the service status of the specified MinIO server
-	_, e := client.ServerInfo(globalContext)
-
-	if e != nil {
-		printMsg(serviceRestartMessage{Status: "failure", Err: e, ServerURL: aliasedURL})
-	} else {
-		printMsg(serviceRestartMessage{Status: "success", ServerURL: aliasedURL})
+	// Print restart progress
+	printProgress := func() {
+		if !globalQuiet && !globalJSON {
+			coloring.Printf(mark)
+		}
 	}
 
-	return nil
+	for {
+		select {
+		case <-globalContext.Done():
+			return globalContext.Err()
+		case <-time.NewTimer(3 * time.Second).C:
+			ctx, cancel := context.WithTimeout(globalContext, 1*time.Second)
+			// Fetch the service status of the specified MinIO server
+			info, e := client.ServerInfo(ctx)
```

Comment:
```suggestion
			info, e := client.ServerInfo(ctx)
```

I always wondered if we can avoid this heavy call instead make it a simpler HTTP request to know if the server is up? like reaching out to the `/cluster` endpoint?  

Or probably introduce another separate API that does a much simpler `objLayer` up or not check?

Reply:
> I always wondered if we can avoid this heavy call instead make it a simpler HTTP request to know if the server is up? like reaching out to the `/cluster` endpoint?
> 
> Or probably introduce another separate API that does a much simpler `objLayer` up or not check?

I would say, when can make ServerInfo light when the cluster is not initialized yet. So in total, there is only one heavy call when the cluster is initialized, and that would be the last call in this command before prompting 'successful restart.' and quitting.